### ISSUE TITLE
Allow configuring git to use current VS Code version for editing commits

### ISSUE THREAD

**Issue by mjbvz** [microsoft/vscode#112608]
Issue Type: <b>Bug</b>

## Problem
We can [configure VS Code as the default git editor](https://code.visualstudio.com/docs/editor/versioncontrol#_vs-code-as-git-editor), but this currently hardcodes the `code` program name.

This means that if you are using the insiders version of VS Code and run a git command in the intergrated terminal, we end up launching  the stable version of VS Code to edit the commit message

## Feature Request
Make it possible to configure git to use the currently running version of VS Code as the default editor, falling back to stable or insiders if the git command was not run from VS Code

VS Code version: Code - Insiders 1.53.0-insider (f3e95ede9e4b9818c29a5bca3564d6fd162fb210, 2020-12-15T08:14:20.855Z)
OS version: Darwin x64 20.1.0

**Comment by Tyriar** [comment]
Do we really want this? The less interactions we have between insiders and stable, we could just mention in the docs that code-insiders can be configured?

**Comment by mjbvz** [comment]
Yes I'd like it, but the VS Code team is probably weird because we regularly use both stable and insiders. If I run a git command in VS Code, I want it to use that version of VS Code

**Comment by connor4312** [comment]
This should be quite doable now with the `environmentVariableCollection` API

**Comment by Tyriar** [comment]
Since git editor config is setup outside VS Code, I still think this should also be set up separately. The way I'd recommend doing this is to creating an alias to be used by git based on whether `$TERM_PROGRAM_VERSION` includes `-insider`. We want to be messing with the user's environment/`$PATH` as little as possible.
